### PR TITLE
Fix for boolean param + PSR2

### PR DESCRIPTION
**Pull request by ooga**
- Fix return param value on 'false'
- Refacto for PSR2 validation

### CODE REVIEW

**Comment by ooga** [comment]
```$parsed_url``` => ```$parsedUrl```
```check_get_params()``` => ```checkSnappyparams()```
set snappy margin options moved into ```checkSnappyparams()```
fix boolean param convertion and moved as new generic function ```checkParam()```

**Comment by Netmisa** [comment]
Is it possible to rebase it please ?

**Comment by vinzentt** [comment]
I think filter_var function is a better option to convert a string param to a boolean.

```php
function convertToBoolean($param) {
    $newParam = filter_var($param, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
    return $newParam === null ? $param : $newParam;
}
```

**Comment by dvdn** [comment]
:recycle: Rebase needed for possible merge

**Comment by ooga** [comment]
Rebase done. About the proposition of using ```filter_var``` on boolean type, i think it's not really usefull. It will be for a more complex validation.

@Netmisa , @dvdn What do you think ?

File: web/index.php [hunk 1]
```diff
@@ -1,90 +1,92 @@
...
+    // Download file instead of viewing it in the browser
+    if (isset($_GET['ddl'])) {
+        $filename = (empty($_GET['ddl']))? 'file' : $_GET['ddl'];
+        header('Content-Disposition: attachment; filename="'.$filename.'.pdf"');
+    }
+
+    // Convert pdf in CMYK colorspace
+    // Need GhostScript
+    // Need a writeable temporary directory for php process
+    if (isset($_GET['cmyk']) && $_GET['cmyk'] === 1) {
```
Comment (dvdn) [inline web/index.php]: Don't we keep '...$_GET['cmyk'] == 1' instead of '=== 1' for the moment ?

**Comment by Netmisa** [comment]
I prefer with "filter_var"

**Comment by vinzentt** [comment]
filter_var does the same (and even more) but in a better way than a rudimentary string comparison IMO.

File: web/index.php (hunk 1 above)
Comment (vinzentt) [inline web/index.php]: with filter_var you could convert it to boolean and avoid this pb.

**Comment by thierrymarianne** [comment]
How do you plan to test these changes? :sweat_smile:

**Comment by thierrymarianne** [comment]
As discussed quickly with @Netmisa  &&  @ooga, we can talk to each other irl between 16:30 and 17:00 today to figure something out ^_

**Pull request merged**